### DISCUSSION TITLE
PyAEDT installation related questions

### DISCUSSION THREAD

**Discussion by Abhi-SimuTech** [ansys/pyaedt#4963] [Q&A] [answered]
Hi,

I have a couple of questions which I would really appreciate your feedback. In the context that someone is running AEDT on tightly-controlled servers that are watched by security software and any program that runs there must be explicitly allowed to run -

1. Do we need admin privileges to run the PyAEDT installation script ("pyaedt_installer_from_aedt.py" from the user Downloads directory? 
2. Where is PyAEDT installed on a Windows box?  Is it a system-wide or user installation?
3. Which "folder" to look for the Jupyter notebook after installation?  
4. What is the procedure uninstall PyAEDT (clean/complete removal) from the system?

Thanks in advance.

Regards

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @Abhi-SimuTech ,

I have helped some users in such kinds of environments, and they usually do not have problems.

The "pyaedt_installer_from_aedt.py" script is doing the following operations:

1. Creates a virtual environment called .pyaedt_venv under appdata directory.
2. Install PyAEDT and all its dependencies (numpy, jupyter...). If the user has IT restrictions to download and install packages, they will need to use the offline installation mode with the wheelhouse, explained here: https://aedt.docs.pyansys.com/version/stable/Getting_started/Installation.html#install-from-pyaedt-installer
3. Finally, it created the 4 buttons in AEDT Automation tab, which is created in the PersonalLib directory, which the user should also has write permissions. You will see how a new directory called "Toolkits" is created, and inside all the directories corresponding to the buttons.

Answering your questions:

1. The script is an Iron Python script that has be to launched from AEDT, it means, that if the user can run any AEDT script, this script should not have any restrictions.

2. The script is installing PyAEDT at the user level, specifically, it creates a virtual environment in %appdata% directory (C:\Users\slopez\AppData\Roaming\.pyaedt_venv), which most the users should have write permissions, even for high-level restricted environments. 

3. In the previously described virtual environment, all PyAEDT dependencies are installed, in addition jupyter and ipython (the console), which are needed to run the 4 buttons installed in AEDT Automation tab.

4. For clean uninstalls, you just need to delete the virtual environment .pyaedt_venv and the "Toolkits" directory in the PersonalLib.

4* If you need to update the PyAEDT release, you just need to rerun the script, it will just update the PyAEDT version to the last one. Of course, if you are an advanced python user, you can go to the virtual environment, and in the terminal update pyaedt (pip install pyaedt -U).

Hopefully, it answers your questions.

Regards,
Samuel

**Reply by Abhi-SimuTech** [reply]
Thanks Samuel, for the very comprehensive reply. Super helpful.
